src/Microsoft.IdentityModel.Tokens/BaseConfigurationComparer.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    internal class BaseConfigurationComparer : IEqualityComparer<BaseConfiguration>
+    {
+        public bool Equals(BaseConfiguration config1, BaseConfiguration config2)
+        {
+            if (config1 == null && config2 == null)
+                return true;
+            else if (config1 == null || config2 == null)
+                return false;
+            else if (config1.Issuer == config2.Issuer && config1.SigningKeys.Count == config2.SigningKeys.Count
+                     && !config1.SigningKeys.Select(x => x.InternalId).Except(config1.SigningKeys.Select(x => x.InternalId)).Any())
+                return true;
+            else
+                return false;
+        }
+
+        public int GetHashCode(BaseConfiguration config)
+        {
+            return (config.Issuer + string.Join("", config.SigningKeys.Select(x => x.InternalId).OrderBy(x => x.FirstOrDefault()))).GetHashCode();
```

Comment:
this might work for the prototype but we shouldn't be doing string operations to produce the hash code in the release version. Can you at least comment what you plan to do here?

Reply:
Here to use issuer and sorted signingKeys.InternalId to generate a hashcode.

Reply:
take a look at something like: [https://stackoverflow.com/questions/59375124/how-to-use-system-hashcode-combine-with-more-than-8-values](https://stackoverflow.com/questions/59375124/how-to-use-system-hashcode-combine-with-more-than-8-values)

We shouldn't be doing string operations in the GetHashCode method.

Reply:
Update the GetHashcode method with "^".

Reply:
Are you doing force pushes here? (I see this changed but it still shows as the first iteration) Once you push a PR, please just push new changes as new commits in case anyone has checked out your branch and is attached to its change history.